cmd/exec.go

```diff
@@ -101,9 +96,22 @@ func executeExec(ctx context.Context, dev *model.Dev, args []string) error {
 		return err
 	}
 
+	if p == nil {
+		return errors.UserError{
```

Comment:
@rberrelleza when working with complex apps, it is usually required to run `okteto exec` to finish the app configuration from files that only exist in the local filesystem. When users face this error, they get really confused.
What could we do? should we run `okteto up` + `okteto exec` + `okteto down` for these cases?

Reply:
@pchico83  I don't understand what you mean here. This error happens if they are trying to do "okteto exec" on a pod that's not on developer mode.   

Are you suggesting that we should allow "okteto exec" on  non-dev pods?

Reply:
The use case is to be able to run `exec` on non-dev pods, but activating dev mode first.
For example, they have a `.env` file not committed in git. The deploy the app, and they want to run:
```
okteto exec -- make configure
```
which uses `.env`. Now, they need to run `okteto up`, `okteto exec` and `okteto down`. Andif they don't do `up`, we show this error, which it is generating confusion.
Maybe it is a use case for `okteto up -- make configure`

Reply:
I totally see this as a use case for the `okteto up -- make configure` or a similar way to execute the command but implying that it is happing in a dev container. Allowing `exec` and magically doing `up` probably causes more confusion about whether it is happening on a non-dev container or not.

Reply:
@rlamana would you automate `okteto down` after `okteto up -- make configure`?

Reply:
so we are talking about a single execution and then it dies and does `okteto down`?

Reply:
I thought of `okteto up -- command` more of a way to do UP and run a command overriding the one defined in the `okteto.yml`. I would not do the `down`.

Reply:
I understand this is kind of a corner case and not an operation you keep doing over and over again. I would focus on keeping everything coherent and intuitive, even if that means to run 2 commands (up and then down), rather than adding more magic that makes it difficult to understand what exactly is happening.

Reply:
> I understand this is kind of a corner case and not an operation you keep doing over and over again. I would focus on keeping everything coherent and intuitive, even if that means to run 2 commands (up and then down), rather than adding more magic that makes it difficult to understand what exactly is happening.

I agree with @rlamana , for me, an "okteto up -- command" means "do an up, and then run `command` instead of the command defined on okteto.yml"